### PR TITLE
Adds forceUpdate to ComponentConfigurationTranslationHelper

### PR DESCRIPTION
**Pull request by oliversarfas**
add boolean to allow for bypas of update and return existing string instead

### CODE REVIEW

**Review by goranambar — CHANGES_REQUESTED** [review]
Hey @oliversarfas I'm afraid this is not what we're looking for. I dropped some directions on how to implement this.

File: src/Translate/Services/TranslateService.php [hunk 1]
```diff
@@ -163,7 +172,9 @@ protected static function addOrUpdateLanguageLine($group, $key, $type, $scenario
             );
         }
 
-        $languageLine->setTranslation($language, $translationString);
+        if ($forceUpdate) {
```
Comment (goranambar) [inline src/Translate/Services/TranslateService.php:175]: So, if `$forceUpdate` is set to `false`, and `LanguageLine` didn't exist previously (as it is created few lines before), it won't set the translation string to it. And we don't want that, right?

File: src/Translate/Services/TranslateService.php [hunk 2]
```diff
@@ -146,10 +147,18 @@ protected static function generateMessageKey($uid, $name): string
...
+        $group,
+        $key,
+        $type,
+        $scenarioId,
+        $language,
+        $translationString,
+        bool $forceUpdate = true,
+    ): string {
         $languageLine = OpenDialogLanguageLine::where('group', $group)->where('key', $key)->first();
         if (!$languageLine) {
```
Comment (goranambar) [inline src/Translate/Services/TranslateService.php:163]: Just check before this line:
`if ($languageLine && !$forceUpdate){return;}`
And delete the `$forceUpdate` check below.

**Review by goranambar — CHANGES_REQUESTED** [review]
Still couple of points to fix.

File: src/Translate/Services/TranslateService.php [hunk 3]
```diff
@@ -86,7 +86,8 @@ public function addOrUpdateCustomTranslation(
         string $name,
         int $scenarioId,
         string $translationString,
-        string $language
+        string $language,
+        bool $forceUpdate,
```
Comment (goranambar) [inline src/Translate/Services/TranslateService.php:90]: This should be not in here but in `addOrUpdateConfigTranslation` right?

File: src/Translate/Services/TranslateServiceInterface.php [hunk 4]
```diff
@@ -103,7 +103,8 @@ public function addOrUpdateCustomTranslation(
         string $name,
         int $scenarioId,
         string $translationString,
-        string $language
+        string $language,
+        bool $forceUpdate,
```
Comment (goranambar) [inline src/Translate/Services/TranslateServiceInterface.php:107]: This should be not in here but in addOrUpdateConfigTranslation right?

**Comment by oliversarfas** [comment]
@goranambar test added and PR feedback actioned

**Review by goranambar — APPROVED** [review]
Looks OK now

**Pull request merged**